x/params/types/common_test.go

```diff
@@ -24,36 +23,78 @@ type params struct {
 	BondDenom     string        `json:"bond_denom" yaml:"bond_denom"`
 }
 
-func validateUnbondingTime(i interface{}) error {
-	v, ok := i.(time.Duration)
+func validateUnbondingTime(currentValue, newValue interface{}) error {
+	newUbdTime, ok := newValue.(time.Duration)
 	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
+		return fmt.Errorf("invalid parameter type: %T", newValue)
 	}
 
-	if v < (24 * time.Hour) {
+	if newUbdTime < (24 * time.Hour) {
 		return fmt.Errorf("unbonding time must be at least one day")
 	}
 
+	// ignore error if there is no previous value
+	if currentValue == nil {
+		return nil
+	}
+
+	currentUbdTime, ok := currentValue.(time.Duration)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", currentValue)
+	}
+
+	if currentUbdTime == newUbdTime {
+		return fmt.Errorf("new unbonding time cannot be the same as the current value %s, %s", currentUbdTime, newUbdTime)
+	}
+
 	return nil
 }
 
-func validateMaxValidators(i interface{}) error {
-	_, ok := i.(uint16)
+func validateMaxValidators(currentValue, newValue interface{}) error {
+	newVals, ok := newValue.(uint16)
 	if !ok {
-		return fmt.Errorf("invalid parameter type: %T", i)
+		return fmt.Errorf("invalid parameter type: %T", newValue)
+	}
+
+	// ignore error if there is no previous value
+	if currentValue == nil {
+		return nil
+	}
+
+	currentVals, ok := currentValue.(uint16)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", currentValue)
+	}
+
+	if currentVals == newVals {
+		return fmt.Errorf("proposed max validators has the same value as the current one, %d = %d", currentVals, newVals)
 	}
 
 	return nil
 }
 
-func validateBondDenom(i interface{}) error {
-	v, ok := i.(string)
+func validateBondDenom(currentValue, newValue interface{}) error {
+	newDenom, ok := newValue.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", newValue)
+	}
+
+	if err := sdk.ValidateDenom(newDenom); err != nil {
+		return err
+	}
+
+	// ignore error if there is no previous value
+	if currentValue == nil {
```

Comment:
I'd put this at the beginning of the function, so that you can avoid the type assertion and the denom validation in case currentValue is nil